### ISSUE TITLE
need to store subscription to make it work

### ISSUE THREAD

**Issue by haifengkao** [kitasuke/SwiftUI-Redux#1]
the call   
`APIService().response(from: SearchRepositoryRequest())`
and
`                action.future
                    .receive(on: DispatchQueue.main)
                    .sink(receiveValue: {
                        next($0)
                    })`

should be 

`subscription = APIService().response(from: SearchRepositoryRequest())`
`                subscription = action.future
                    .receive(on: DispatchQueue.main)
                    .sink(receiveValue: {
                        next($0)
                    })`

if you don't store the subscription somewhere, the subscription will be cancelled immediately.

**Comment by kitasuke** [comment]
That's true. I guess I haven't updated since it worked without storing subscription, will update them. Thanks!
